components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/TenantAwareAxis2ConfigurationContextObserver.java

```diff
@@ -0,0 +1,238 @@
+                if (resourceFile != null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("File for publisher name: " + resourceFile.getName()
+                                + " is available in the configuration " + "store.");
+                    }
+                    getResourceManager().addEventPublisherConfiguration(resourceFile);
+                }
+            }
+        } catch (ConfigurationManagementException e) {
+            if (e.getErrorCode()
+                    .equals(ConfigurationConstants.ErrorMessages.ERROR_CODE_FEATURE_NOT_ENABLED.getCode())) {
+                log.warn("Configuration store is disabled. Super tenant configuration will be used for the tenant "
+                        + "domain: " + PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+            } else {
+                log.error(populateMessageWithData(ERROR_CODE_ERROR_WHEN_FETCHING_TENANT_SPECIFIC_PUBLISHER_FILES,
+                        PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain()), e);
+            }
+        } catch (TenantResourceManagementException e) {
+            log.error(populateMessageWithData(ERROR_CODE_ERROR_WHEN_ADDING_EVENT_PUBLISHER_CONFIGURATION,
+                    PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain()), e);
+        }
+    }
+
+    /**
+     * This method returns super tenant event publisher configurations.
+     *
+     * @return list of event publisher configurations.
+     */
+    private List<EventPublisherConfiguration> getSuperTenantEventPublisherConfigurations() {
+
+        List<EventPublisherConfiguration> activeEventPublisherConfigurations = null;
+        try {
+            activeEventPublisherConfigurations = TenantResourceManagerDataHolder.getInstance()
+                    .getEventPublisherService().getAllActiveEventPublisherConfigurations();
+        } catch (EventPublisherConfigurationException e) {
+            log.error(populateMessageWithData(
+                    TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_FETCHING_SUPER_TENANT_EVENT_PUBLISHER_CONFIGURATION,
+                    PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain()), e);
+        }
+        return activeEventPublisherConfigurations;
+    }
+
+    /**
+     * This method returns super tenant event stream configurations.
+     *
+     * @return list of event stream configurations.
+     */
+    private List<EventStreamConfiguration> getSuperTenantEventStreamConfigurations() {
+
+        List<EventStreamConfiguration> eventStreamConfigurationList = null;
+        try {
+            eventStreamConfigurationList = TenantResourceManagerDataHolder.getInstance().getCarbonEventStreamService()
+                    .getAllEventStreamConfigurations();
+        } catch (EventStreamConfigurationException e) {
+            log.error(handleServerException(
```

Comment:
Exception conversion is not needed right

Reply:
fixed